src/configuration.ts

```diff
@@ -94,60 +98,86 @@ export class ConnectionConfiguration implements IConnectionConfiguration {
     return p;
   }
 
-  get sslCertDir():string{
+  get sslCertDir(): string {
     return path.join(this.puppetDir, 'ssl', 'certs');
   }
 
-  get sslCertFile():string{
+  get sslCertFile(): string {
     return path.join(this.puppetDir, 'ssl', 'cert.pem');
   }
 
-  get environmentPath():string{
-    return path.join(this.puppetDir, 'bin') + this.pathEnvSeparator() +
-      path.join(this.facterDir, 'bin') + this.pathEnvSeparator() +
-      path.join(this.hieraDir, 'bin') + this.pathEnvSeparator() +
-      path.join(this.mcoDir, 'bin') + this.pathEnvSeparator() +
-      path.join(this.puppetAgentDir, 'bin') + this.pathEnvSeparator() +
-      path.join(this.rubydir, 'bin') + this.pathEnvSeparator() +
+  get environmentPath(): string {
+    return (
+      path.join(this.puppetDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.facterDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.hieraDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.mcoDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.puppetAgentDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.rubydir, 'bin') +
+      this.pathEnvSeparator() +
       path.join(this.puppetAgentDir, 'sys', 'tools', 'bin') +
-      this.pathEnvSeparator();
+      this.pathEnvSeparator()
+    );
   }
 
-  get languageServerPath():string{
+  get languageServerPath(): string {
     return this.context.asAbsolutePath(path.join('vendor', 'languageserver', 'puppet-languageserver'));
   }
 
-  get type(): ConnectionType{
-    if (this.host === '127.0.0.1' ||
-      this.host === 'localhost' ||
-      this.host === '') {
+  get type(): ConnectionType {
+    if (this.host === '127.0.0.1' || this.host === 'localhost' || this.host === '') {
       return ConnectionType.Local;
     } else {
       return ConnectionType.Remote;
     }
   }
 
-  get languageServerCommandLine():Array<string>{
+  get protocol(): ProtocolType {
+    switch (this.config['languageclient']['protocol']) {
+      case 'stdio':
+        return ProtocolType.STDIO;
+      case 'tcp':
+        return ProtocolType.TCP;
+      default:
+        return ProtocolType.STDIO;
+    }
+  }
+
+  get languageServerCommandLine(): Array<string> {
     var args = new Array<string>();
 
-    if ((this.host === undefined) || (this.host === '')) {
-      args.push('--ip=127.0.0.1');
-    } else {
-      args.push('--ip=' + this.host);
+    switch (this.protocol) {
+      case ProtocolType.STDIO:
+        args.push('--stdio');
+        break;
+      case ProtocolType.TCP:
+        if (this.host === undefined || this.host === '') {
+          args.push('--ip=127.0.0.1');
```

Comment:
might be best to change this to `localhost`.  Supports ip6 and ip4 then

Reply:
in a PR about STDIO?

Reply:
whoops....True...just saw the green